### PR TITLE
Payment API: decorateDemand endpoint

### PR DESCRIPTION
**Pull request by Wiezzel**
Generate payment-related properties and constraints to be added to a demand published on the marketplace. As a parameter it accepts a list of IDs of allocations to be used to pay for invoices resulting from the decorated demand.

### CODE REVIEW

File: specs/payment-api.yaml [hunk 1]
```diff
@@ -1445,3 +1486,30 @@ components:
       required:
         - platform
         - address
+
+    MarketDecoration:
```
Comment (tworec) [inline specs/payment-api.yaml:1490]: There is `DemandOfferBase` defined in Market API. Consider reusing it. We can move it to `common.yaml` if needed.

**Review by filipgolem — APPROVED** [review]
Looks good to me.

File: specs/payment-api.yaml (hunk 1 above)
Comment (Wiezzel) [reply specs/payment-api.yaml:1490]: @tworec I agree that re-using a common base class would be a good idea, but this PR is really urgent and I would like to merge it ASAP. Also, I am not quite convinced which design is better – serialized dictionary (JSON–in–JSON) vs. explicit key-value list. Therefore, please forgive me if I merge this without applying your suggestion.

**Review by maaktweluit — APPROVED** [review]

File: specs/payment-api.yaml (hunk 1 above)
Comment (tworec) [reply specs/payment-api.yaml:1490]: no worries

**Pull request merged**